### ISSUE TITLE
unshallow() doesn't unshallow the repo

### ISSUE THREAD

**Issue by botcheddevil** [semantic-release/semantic-release#764]
# Bug report

## Current behavior

unshallow should unshallow doesnt work as $GIT_CREDENTIALS are not provided to it, 

- which fails the 'commit analyser' to make next version number and it defaults to v1.0.0
- thus pushing to the origin repo causes a conflict between tags, as v1.0.0 preexists.

## Expected behavior

`unshallow()` should unshallow the repository, which in turn should allow 'commit analyzer' to make next version number and then push to the origin repo.

### Possible solution

Provide `unshallow()` with `origin` param

#763

## Environment

- GIT hosting provider: **Bitbucket.org**
- **semantic-release** version: **15.1.7**
- CI environment: **Jenkins**
- Plugins used: **exec**
- **semantic-release** configuration: 

```{
      "branch": "master",
      "verifyConditions": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Conditions verified'"
        }
      ],
      "analyzeCommits": {
        "preset": "angular"
      },
      "verifyRelease": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Release verified'"
        }
      ],
      "generateNotes": [
        "@semantic-release/release-notes-generator"
      ],
      "prepare": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Prepare'"
        }
      ],
      "publish": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Published'"
        }
      ],
      "success": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Success'"
        }
      ],
      "fail": [
        {
          "path": "@semantic-release/exec",
          "cmd": "echo 'Failure'"
        }
      ]
}
```

```
[Semantic release]: Create tag v1.0.0
[Semantic release]: An error occurred while running semantic-release: { Error: Command failed: git tag v1.0.0
fatal: tag 'v1.0.0' already exists


    at makeError (/usr/local/node/lib/node_modules/semantic-release/node_modules/execa/index.js:172:9)
    at Promise.all.then.arr (/usr/local/node/lib/node_modules/semantic-release/node_modules/execa/index.js:277:16)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:118:7)
  code: 128,
  stdout: '',
  stderr: 'fatal: tag \'v1.0.0\' already exists\n',
  failed: true,
  signal: null,
  cmd: 'git tag v1.0.0',
  timedOut: false,
  killed: false }
{ Error: Command failed: git tag v1.0.0
fatal: tag 'v1.0.0' already exists


    at makeError (/usr/local/node/lib/node_modules/semantic-release/node_modules/execa/index.js:172:9)
    at Promise.all.then.arr (/usr/local/node/lib/node_modules/semantic-release/node_modules/execa/index.js:277:16)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:118:7)
  code: 128,
  stdout: '',
  stderr: 'fatal: tag \'v1.0.0\' already exists\n',
  failed: true,
  signal: null,
  cmd: 'git tag v1.0.0',
  timedOut: false,
  killed: false }

```

**Comment by pvdlg** [comment]
Are you sure the problem is with `unshallow()` ?
If your CI managed to clone the repo, then `git fetch --unshallow -- --tags` should work without any additional authentication mechanism than what's provided by the CI.

What CI are you using? 
What do you get if you run `git fetch --unshallow -- --tags` on your CI?

**Comment by botcheddevil** [comment]
Hi @pvdlg ,

Thank you for your response. 

> Are you sure the problem is with unshallow() ?

Yes! check my pull-request it solves my problem. 

#763 (all unit tests pass except unshallow one as its expecting a parameter now) 

> What CI are you using?

I am using Jenkins with pipeline scripts's module checkout scm. Based on your response I presume you expect Jenkins to store credentials in 'origin' of .git/config??

> What do you get if you run git fetch --unshallow -- --tags on your CI?

On the bash I get prompted for the username & password.

**Comment by pvdlg** [comment]
> I am using Jenkins with pipeline scripts's module checkout scm. Based on your response I presume you expect Jenkins to store credentials in 'origin' of .git/config??

Not particularly. I don't use Jenkins and I'm not really sure how it works regarding credentials. I know there is other people using Jenkins and they don't experience this issue.
We can add the credentials to the unshallow command though.

**Comment by botcheddevil** [comment]
#766 PR seems to be passing all tests now.

> I know there is other people using Jenkins and they don't experience this issue.

Could you refer me to someone who uses Jenkins with s-r, not sure what am I missing? Most people like using ssh where such use cases will automatically pass.


On an unrelated note, why do we need to `unshallow`? Could we use 

`git ls-remote --tags /url/to/upstream/repo`

`unshallow` completely defeats the purpose of having a shallow repo, now that I think of it, if I configured Jenkins for a full depth clone I wouldn't have got into this mess in the first place.

**Comment by pvdlg** [comment]
Regarding people using Jenkins you can ask on the Gitter channel.

We use unshallow because we need both the tags and the commit messages as semantic-release analyze those.
`git ls-remote` would give us only refs (commit sha and tags).

**Comment by pvdlg** [comment]
Fixed in #766
